src/session.ts

```diff
@@ -303,6 +305,26 @@ export class SessionManager implements Middleware {
         await this.start();
     }
 
+    /** In Development mode, write the PID to a file where the parent session can find it, to attach the dotnet debugger. */
+    private async writePidIfInDevMode(pwshProcess: PowerShellProcess): Promise<void> {
+        if (this.extensionContext.extensionMode !== vscode.ExtensionMode.Development) { return; }
+        const parentSessionId = process.env.VSCODE_PARENT_SESSION_ID;
+        const pidFilePath = vscode.Uri.joinPath(this.sessionsFolder, `PSES-${parentSessionId}.pid`);
+
+        if (parentSessionId === undefined) { return; }
+
+        const fs = vscode.workspace.fs;
+        const pid = (await pwshProcess.getPid())!.toString();
+        await fs.writeFile(pidFilePath, Buffer.from(pid));
+        const deletePidOnExit = pwshProcess.onExited(() => {
+            deletePidOnExit.dispose();
+            fs.delete(pidFilePath, {useTrash: false});
```

Comment:
Should we be setting `useTrash` for some other internal file deletions too?

Reply:
I think it makes sure it doesn't go to the recycle bin so probably, it's not super clear in the documentation and I haven't delved the actual code.